tests/phpunit/Unit/PropertyAnnotators/UserEditCountPerNsPropertyAnnotatorTest.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+namespace SESP\Tests\PropertyAnnotators;
+
+use SESP\PropertyAnnotators\UserEditCountPerNsPropertyAnnotator;
+use SMW\DIProperty;
+use SMW\DIWikiPage;
+
+/**
+ * @covers \SESP\PropertyAnnotators\UserEditCountPerNsPropertyAnnotator
+ * @group semantic-extra-special-properties
+ *
+ * @license GNU GPL v2+
+ * @since 2.0
+ *
+ * @author Alexander Mashin
+ */
+class UserEditCountPerNsPropertyAnnotatorTest extends \PHPUnit_Framework_TestCase {
+
+	private $property;
+	private $appFactory;
+
+	protected function setUp() {
+		parent::setUp();
+
+		$this->appFactory = $this->getMockBuilder( '\SESP\AppFactory' )
+			->disableOriginalConstructor()
+			->getMock();
+
+		$this->property = new DIProperty( '___USEREDITCNTNS' );
+	}
+
+	public function testCanConstruct() {
+
+		$this->assertInstanceOf(
+			UserEditCountPerNsPropertyAnnotator::class,
+			new UserEditCountPerNSPropertyAnnotator( $this->appFactory )
+		);
+	}
+
+	public function testIsAnnotatorFor() {
+
+		$instance = new UserEditCountPerNsPropertyAnnotator(
+			$this->appFactory
+		);
+
+		$this->assertTrue(
+			$instance->isAnnotatorFor( $this->property )
+		);
+	}
+
+	/**
+	 * @dataProvider editCountPerNsProvider
+	 */
+	public function testGetEditsPerNs() {
+
+	}
+
+	/**
+	 * @dataProvider editCountPerNsProvider
+	 */
+	public function testAddAnnotation( $count, $expected ) {
+
+		$user = $this->getMockBuilder( '\User' )
+			->disableOriginalConstructor()
+			->getMock();
+
+		$user->expects( $this->once() )
+			->method( 'getEditsPerNs' )->willReturn( [ 0 => $count ] );
+
+		$this->appFactory->expects( $this->once() )
+			->method( 'newUserFromTitle' )
+			->will( $this->returnValue( $user ) );
```

Comment:
This is a lot of setup code with many details. Try extract

$user = $this->newWhateverUser();
$stuff = $this->anotherPrivateMethod();

Reply:
Mr @JeroenDeDauw, thank you for the code review. I have closed this pull request because I rewrote the unit test completely.